Migrate: Postgres

melon095 opened the pull request:
This pull request aims to fully migrate the bot over to postgres.

**How to automatically migrate tables**
Use [PGLoader](https://github.com/dimitri/pgloader)
Will call the table `twitch`, rename to `bot`
Migration table -- update version to 2

Closes #8

melon095 committed:
Add postgres as a package

melon095 committed:
Refactor(Postgres): Migrations

melon095 committed:
Move to postgres.

Also prettier.rc longer width

melon095 committed:
Merge branch 'main' into migrate-postgres

melon095 commented on package.json:
```diff
@@ -62,6 +62,7 @@
 		"got": "^12.0.2",
 		"humanize-duration": "^3.27.1",
 		"mysql2": "^2.3.3",
```
Remove this

melon095 commented on src/controller/Banphrase/index.ts:
```diff
@@ -43,27 +43,21 @@ export class Banphrase {
-			[this.Name],
-		)
-			.then((_p) => _p.ArrayOrNull())
+		Bot.SQL.Query<PhraseType[]>`SELECT Phrase FROM banphrases WHERE channel = ${this.Name}`
```
phrase

melon095 commented on src/controller/Banphrase/index.ts:
```diff
@@ -43,27 +43,21 @@ export class Banphrase {
-				}
-			});
+		Bot.SQL.Query<Database.banphrases[]>`
+            SELECT Phrase 
```
phrase

melon095 commented on src/controller/Commands/Handler.ts:
```diff
@@ -98,12 +88,7 @@ export class CommandsHandler {
-								'build/commands',
-								command.name,
-							);
+							'file://' + resolve(process.cwd(), 'build/commands', command.name);
```
tools.Import

melon095 commented on src/controller/Commands/Handler.ts:
```diff
@@ -41,35 +41,25 @@ export class CommandsHandler {
-			const name =
-				'file://' +
-				resolve(process.cwd(), 'build/commands', command.Name + '.js');
+			const name = 'file://' + resolve(process.cwd(), 'build/commands', command.Name + '.js');
```
tools.Import

melon095 commented on src/controller/DB/index.ts:
```diff
@@ -1,223 +1,73 @@
+        `.then(async ([row]) => {
+		if (row === undefined) {
+			await createDefaultMigrationTable(db);
+			db.Query`INSERT INTO migration (version) VALUES (0)`;
```
await

melon095 commented on src/controller/DB/index.ts:
```diff
@@ -226,39 +76,36 @@ export class SQLController {
-				.split(';')
-				.map((query) => query.trim())
-				.filter(Boolean);
+			// const migration = resolve(process.cwd(), 'Migrations', `${version}_${name}`);
```
Remove comments

melon095 commented on src/loops/loops.ts:
```diff
@@ -101,9 +88,10 @@ export function __loops() {
-						Bot.SQL.query(
-							'UPDATE `channels` SET `viewers` = JSON_ARRAY(?) WHERE `name` = ?',
-							[viewers.split(','), name],
+						Bot.SQL.Query(
```
Parenthesis

melon095 committed:
Fix(Postgres): Review Comments

melon095 committed:
Merge branch 'main' into migrate-postgres

Commit:
Formatted by Prettier in CI

melon095 reviewed (commented):
Ok

melon095 committed:
Too lazy to make a message

melon095 committed:
Merge branch 'migrate-postgres' of https://github.com/JoachimFlottorp/Melonbot into migrate-postgres

melon095 reviewed (commented):
Ok

hotbear1110 commented on config.example.json:
```diff
@@ -5,9 +5,7 @@
-		"Host": "",
-		"User": "",
-		"Password": ""
+		"Address": "postgres://user:password@localhost:5432/melonbot"
```
Hmm doesn't this depend on how the user has setup postgres?

melon095 replied:
Nah incase theres no password setup, im fairly sure you just omit it. So: `postgres://user@localhost:5432/foo`

hotbear1110 replied:
eShrug ok

hotbear1110 reviewed (approved):
Seems good I guess. Clueless not like I can actually test anything

melon095 committed:
Fix: Accidently joining bot channel twice

melon095 committed:
Migration: BigInts to Int

BigInts are casted to string, so just moving it to Int, as the highest number is 4
